driver-core/src/test/functional/com/mongodb/internal/connection/DefaultConnectionPoolTest.java

```diff
@@ -443,6 +443,41 @@ public void checkoutHandOverMechanism() throws InterruptedException, TimeoutExce
         }
     }
 
+    @Test
+    public void invalidateAfterCloseMustNotThrow() {
+        provider = new DefaultConnectionPool(
+                SERVER_ID,
+                connectionFactory,
+                ConnectionPoolSettings.builder().maxSize(1).build(),
+                mockSdamProvider());
+        provider.ready();
+        provider.close();
+        provider.invalidate(null);
+    }
+
+    @Test
+    public void invalidateConcurrentWithCloseMustNotThrow() throws ExecutionException, InterruptedException {
```

Comment:
I don't understand how this method tests that both start _and_ runOnceAndStop wrap their calls to execute in `doIgnoringRejectedExecution`.  I proved to myself that it does (since if I comment out either one the test fails), but I don't understand why it's the case.  Can you explain?

Reply:
`BackgroundMaintenanceManager.start` is called by `DefaultConnectionPool.ready`, and `BackgroundMaintenanceManager.runOnceAndStop` is called by `DefaultConnectionPool.invalidate`. However, this is not a very important implementation detail. The `invalidateConcurrentWithCloseMustNotThrow` test checks that neither `ready` nor `invalidate` throws an exception when are called concurrently with `close`, because this is the behavior we care about. It does so by running them concurrently and checking that `ready`/`invalidate` do not throw by calling `backgroundTask.get()`. I should rename `backgroundTask` to `readyAndInvalidateResult`.